classes/webservice.php

```diff
@@ -461,12 +468,23 @@ private function get_least_recently_active_paid_user_id() {
         $userslist = $this->list_users();
 
         foreach ($userslist as $user) {
-            if ($user->type != ZOOM_USER_TYPE_BASIC && isset($user->last_login_time)) {
-                // Count the user if we're including all users or if the user is on this instance.
-                if (!$this->instanceusers || core_user::get_user_by_email($user->email)) {
-                    $usertimes[$user->id] = strtotime($user->last_login_time);
-                }
+            // Skip Basic user accounts.
+            if ($user->type == ZOOM_USER_TYPE_BASIC) {
+                continue;
+            }
+            // Skip the protected group.
+            if (!empty($this->protectedgroups) && array_intersect($this->protectedgroups, $user->group_ids ?? [])) {
+                continue;
+            }
+            // We need the login time.
+            if (!isset($user->last_login_time)) {
+                continue;
             }
+            // Count the user only if we're including all users or if the user is on this instance.
```

Comment:
I recommend blank lines before comments and between code blocks for readability.
```suggestion
            // Skip Basic user accounts.
            if ($user->type == ZOOM_USER_TYPE_BASIC) {
                continue;
            }

            // Skip the protected group.
            if (!empty($this->protectedgroups) && array_intersect($this->protectedgroups, $user->group_ids ?? [])) {
                continue;
            }

            // We need the login time.
            if (!isset($user->last_login_time)) {
                continue;
            }

            // Count the user only if we're including all users or if the user is on this instance.
```